packages/design-system/src/components/Tooltip/TooltipTrigger.tsx

```diff
@@ -80,20 +80,35 @@ export const TooltipTrigger: React.FC<TooltipTriggerProps> = ({
   });
 
   let frame: number;
+  const pointerOffset = 15;
   const updateTooltipPosition = (x: number, y: number) => {
     if (typeof frame !== "undefined") {
       window.cancelAnimationFrame(frame);
     }
+
+    let offsetLeft = x;
+    // if tooltip doesn't have enough space on the right move it to the left relative to pointer. Works only when maxWidth is specified
+    if (
+      maxWidth &&
```

Comment:
at a glance we only seem to use this option about half the time in practice. Is there some reasonable way around this limitation? Maybe we consider pulling in a library that solves this, such as [Floating UI](https://floating-ui.com/docs/react-dom), rather than trying to reinvent the wheel if it's not straightforward

Reply:
@nasaownsky I like Ian's suggestion to look into a library that has solved this positioning problem. Were you able to look into Floating UI?

Reply:
@macfarlandian @jovergaag Yeah, I looked into it, but can't get it to work. `x` and `y`  isn't updating somehow, even though I followed the docs impeccably. 

Answering Ian's question about `maxWidth` limitation: when the tooltip content is very large, it automatically expands to the right edge of the window and updates its width on mouse movement, becoming narrower to infinity. And if we put some limiter, for example 300, the tooltip will change its position to the left relative to the pointer correctly, but since this logic still work the tooltip will continue to expand towards the right edge. With `maxWidth` such problem doesn't exist. 

Anyway, I pushed up the commit with updated logic. I would glad if you test it and leave me your thought & suggestions here!